consensus/ssz/src/decode/impls.rs

```diff
@@ -433,35 +484,27 @@ pub fn decode_list_of_variable_length_items<T: Decode>(
         )));
     }
 
-    // Only initialize the vec with a capacity if a maximum length is provided.
-    //
-    // We assume that if a max length is provided then the application is able to handle an
-    // allocation of this size.
-    let mut values = if max_len.is_some() {
```

Comment:
Ooh good point. I kind of hoped the magic of `collect` and size hints would let us keep the pre-allocation, but it seems it only works if the iterator implements `TrustedLen`: https://github.com/rust-lang/rust/blob/48316dfea1914d25189fa441e7310449ed76a446/library/alloc/src/vec/spec_extend.rs#L36. And the iterator from `process_results` doesn't (because it can fail).

It would be quite straight-forward to add a `new_with_capacity` function to `TryCollect` if it turns out we need it

Reply:
Or we could just change the implementation for `Vec` so that it always uses the `size_hint`